File: app/src/main/java/hu/suppoze/mitigyunkma/activity/MainActivity.java
package hu.suppoze.mitigyunkma.activity;

import android.support.design.widget.AppBarLayout;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewCompat;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import butterknife.ButterKnife;
import butterknife.InjectView;
import hu.suppoze.mitigyunkma.R;
import hu.suppoze.mitigyunkma.view.adapter.MainPagerAdapter;

public class MainActivity extends AppCompatActivity {

    @InjectView(R.id.activity_main_appbar) AppBarLayout appBarLayout;
    @InjectView(R.id.activity_main_toolbar) Toolbar toolbar;
    @InjectView(R.id.activity_main_tablayout) TabLayout tabLayout;
    @InjectView(R.id.activity_main_viewpager) ViewPager viewPager;

    private static final float TOOLBAR_ELEVATION = 2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.inject(this);

        initializeActionBar();
        initializeTabLayout();
        initializeViewPager();
        setListenerForTabLayout();
    }

    private void initializeActionBar() {
        setSupportActionBar(toolbar);
        ViewCompat.setElevation(appBarLayout, TOOLBAR_ELEVATION);
    }

    private void initializeTabLayout() {
        tabLayout.addTab(tabLayout.newTab().setText("Tab 1"));
        tabLayout.addTab(tabLayout.newTab().setText("Tab 2"));
        tabLayout.addTab(tabLayout.newTab().setText("Tab 3"));
        tabLayout.setTabGravity(TabLayout.GRAVITY_FILL);
    }

    private void initializeViewPager() {
        final MainPagerAdapter adapter = new MainPagerAdapter(getSupportFragmentManager());
        viewPager.setAdapter(adapter);
        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
    }

    private void setListenerForTabLayout() {
        tabLayout.setOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                viewPager.setCurrentItem(tab.getPosition());
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {}
            @Override
            public void onTabReselected(TabLayout.Tab tab) {}
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onPrepareOptionsMenu(Menu menu) {
        return super.onPrepareOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/hu/suppoze/mitigyunkma/usecase/MainActivity.kt
package hu.suppoze.mitigyunkma.usecase

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import hu.suppoze.mitigyunkma.R
import hu.suppoze.mitigyunkma.extension.setIconColorStateList
import kotlinx.android.synthetic.main.activity_main.*
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initializeActionBar()
        initializeTabLayout()
        initializeViewPager()
    }

    override fun onStart() {
        EventBus.getDefault().register(this)
        super.onStart()
    }

    override fun onStop() {
        EventBus.getDefault().unregister(this)
        super.onStop()
    }

    fun refreshTitle() {
        title = mainActivityViewpager.adapter.getPageTitle(mainActivityViewpager.currentItem)
    }

    private fun initializeActionBar() {
        setSupportActionBar(mainActivityToolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(false)
    }

    private fun initializeTabLayout() {
        mainActivityTabLayout.tabGravity = TabLayout.GRAVITY_FILL
        mainActivityTabLayout.setIconColorStateList(this, R.color.selector_tablayout_icon)
        setListenerForTabLayout()
    }

    private fun initializeViewPager() {
        val adapter = MainPagerAdapter(supportFragmentManager)
        mainActivityViewpager.adapter = adapter
        mainActivityViewpager.offscreenPageLimit = adapter.count
        mainActivityViewpager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(mainActivityTabLayout))
        setListenerForViewPager()
    }

    private fun setListenerForTabLayout() {
        mainActivityTabLayout.setOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                mainActivityViewpager.currentItem = tab.position
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {
            }

            override fun onTabReselected(tab: TabLayout.Tab) {
            }
        })
    }

    private fun setListenerForViewPager() {
        val onPageChangeListener = object : ViewPager.OnPageChangeListener {
            override fun onPageSelected(position: Int) {
                if (position == MainPagerAdapter.Page.CALCULATE.ordinal) {
                    mainActivityAppbar.setExpanded(true, true)
                }
                title = mainActivityViewpager.adapter.getPageTitle(position)
            }

            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
            }

            override fun onPageScrollStateChanged(state: Int) {
            }
        }
        mainActivityViewpager.addOnPageChangeListener(onPageChangeListener)
        mainActivityViewpager.post { onPageChangeListener.onPageSelected(MainPagerAdapter.Page.CALCULATE.ordinal) }
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onNavigateToPage(event: NavigateToPageEvent) {
        mainActivityViewpager.currentItem = event.page.ordinal
    }
}